### PR TITLE
[MGDOBR-420] Generate TS SDK from OpenApi spec

### PR DESCRIPTION
**Pull request by kelvah**
https://issues.redhat.com/browse/MGDOBR-420

This PR is setting up scripts to generate the OpenAPI TS SDK using [openapi-generator-cli](https://github.com/OpenAPITools/openapi-generator-cli), similarly to what is done in the [app services SDK repo](https://github.com/redhat-developer/app-services-sdk-js).

The smart events TS SDK is located in the `openapi/generated` folder.

There are two new npm scripts:
* `npm run fetch-openapi` fetches the spec file from the `sandbox` repo. The file is placed inside `openapi/smart-events.yml`.
* `npm run generate-openapi` generates a new SDK from the spec file into `openapi/generated`

### CODE REVIEW

**Review by vpellegrino — APPROVED** [review]
@kelvah PR looks fine for me.
I agree with your considerations about improving open API annotations for providing better-generated models.

One general observation is: why do we prefer to use `.sh` scripts instead, for example, of node scripts?
I'm asking it just because, by using `.sh` scripts, we are reducing the supported platforms (windows OS will be excluded) that will be able to download new open API spec and generate the client.

**Comment by kelvah** [comment]
> @kelvah PR looks fine for me. I agree with your considerations about improving open API annotations for providing better-generated models.
> 
> One general observation is: why do we prefer to use `.sh` scripts instead, for example, of node scripts? I'm asking it just because, by using `.sh` scripts, we are reducing the supported platforms (windows OS will be excluded) that will be able to download new open API spec and generate the client.

You have a point.

I have blatantly reused the scripts I found in the app services repo. At some point our SDK will be properly generated there, so I didn't spent too much time in setting up this PR.

**Review by jomarko — COMMENTED** [review]
Thank you for a PR @kelvah . I have some proposals to make scripts runnable more ways. Please see https://github.com/kelvah/sandbox-ui/pull/1.

File: README.md [hunk 1]
```diff
@@ -27,3 +27,18 @@ npm run start:federate
...
+
+## OpenAPI TS SDK 
+
+The OpenAPI SDK is located in the `openapi/generated` directory.
+
+It's possible to generate a fresh SDK by running:
+
+```sh
+npm run fetch-openapi
+npm run generate-openapi
```
[the remark this answers is not in the document]
Comment (kelvah) [reply README.md:39]: I don't remember seeing the confirmation prompt while working on it.
Which npm version are you using?

[the remark this answers is not in the document]
Comment (jomarko) [reply README.md:39]: @kelvah `npm -v` says `8.1.2`

[the remark this answers is not in the document]
Comment (kelvah) [reply README.md:39]: Thank you. Strange. Anyway, I see there's a [`--yes` param](https://docs.npmjs.com/cli/v8/commands/npx#compatibility-with-older-npx-versions) to override the prompt. I will add it.

[the remark this answers is not in the document]
Comment (kelvah) [reply README.md:39]: Done with [789500f](https://github.com/kelvah/sandbox-ui/commit/789500f927498e14786feb389374c39b878cb782)

**Review by jomarko — APPROVED** [review]
Thank you @kelvah !

**Review by manstis — APPROVED** [review]
LGTM :+1:

**Pull request merged**